src/main/java/spring/boot/app/demo/controller/UserController.java

```diff
@@ -1,23 +1,35 @@
 package spring.boot.app.demo.controller;
 
 import java.util.List;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import spring.boot.app.demo.model.User;
 import spring.boot.app.demo.service.UserService;
 
 @RestController
 @RequestMapping(value = "/user")
 public class UserController {
     private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
 
-    public UserController(UserService userService) {
+    public UserController(UserService userService, PasswordEncoder passwordEncoder) {
         this.userService = userService;
+        this.passwordEncoder = passwordEncoder;
     }
 
     @GetMapping(value = "/most-active")
     public List<String> getMostActiveUsers(@RequestParam int limit) {
         return userService.getMostActiveLimitedTo(limit);
     }
+
+    @PostMapping(value = "/register")
```

Comment:
```suggestion
    @PostMapping(value = "/")
```
по ідеї в тебе тут просто метод add, і відповідно до конвеншина це зрозуміло буде з того, що тут Post запит, тому register можна прибрати